controllers/cinderapi_controller.go

```diff
@@ -253,20 +253,11 @@ func (r *CinderAPIReconciler) reconcileDelete(ctx context.Context, instance *cin
 	// It's possible to get here before the endpoints have been set in the status, so check for this
 	if instance.Status.APIEndpoints != nil {
 		for _, ksSvc := range keystoneServices {
-			ksSvcSpec := keystonev1.KeystoneServiceSpec{
-				ServiceType:        ksSvc["type"],
-				ServiceName:        ksSvc["name"],
-				ServiceDescription: ksSvc["desc"],
-				Enabled:            true,
-				APIEndpoints:       instance.Status.APIEndpoints[ksSvc["name"]],
-				ServiceUser:        instance.Spec.ServiceUser,
-				Secret:             instance.Spec.Secret,
-				PasswordSelector:   instance.Spec.PasswordSelectors.Service,
-			}
-
-			ksSvcObj := keystone.NewKeystoneService(ksSvcSpec, instance.Namespace, map[string]string{}, 10)
 
+			ks, _ := keystone.GetKeystoneServiceWithName(ctx, helper, ksSvc["name"], instance.Namespace)
```

Comment:
This has the downside of having additional requests, because we are getting the `KeystoneService` CR instead of building the spec ourselves, but it's cleaner.

We should check that the service was actually found and we didn't get an error (renaming `_` to `err` and checking if it's not `nil`), if it's an error and it's not found (`if k8s_errors.IsNotFound(err) {`) then we can assume it was deleted, otherwise we must fail.

Reply:
Yeah, makes sense, we can return earlier if the object isn't there `if k8s_errors.IsNotFound(err) {` and we must fail if `err != nil` because we should catch problematic status here.
I'll try to push an update with this logic.